blockchain/state_transition.go

```diff
@@ -347,11 +358,72 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 
 	rules := st.evm.ChainConfig().Rules(st.evm.Context.BlockNumber)
 
+	// Verify authorization list is not empty.
+	if msg.AuthorizationList() != nil && len(msg.AuthorizationList()) == 0 {
+		return nil, fmt.Errorf("%w: address %v", ErrEmptyAuthList, msg.ValidatedSender())
+	}
+
 	// Execute the preparatory steps for state transition which includes:
 	// - prepare accessList(post-berlin)
 	// - reset transient storage(eip 1153)
 	st.state.Prepare(rules, msg.ValidatedSender(), msg.ValidatedFeePayer(), st.evm.Context.Coinbase, msg.To(), vm.ActivePrecompiles(rules), msg.AccessList())
 
+	// Check authorizations list validity.
+	if msg.AuthorizationList() != nil {
+		for _, auth := range msg.AuthorizationList() {
+			// Verify chain ID is 0 or equal to current chain ID.
+			if auth.ChainID != big.NewInt(0) && st.evm.ChainConfig().ChainID != auth.ChainID {
+				continue
+			}
+			// Limit nonce to 2^64-1 per EIP-2681.
+			if auth.Nonce+1 < auth.Nonce {
+				continue
+			}
+			// Validate signature values and recover authority.
+			authority, err := auth.Authority()
+			if err != nil {
+				continue
+			}
+			// Check the authority account 1) doesn't have code or has exisiting
+			// delegation 2) matches the auth's nonce
+			st.state.AddAddressToAccessList(authority)
+			code := st.state.GetCode(authority)
+			if _, ok := types.ParseDelegation(code); len(code) != 0 && !ok {
+				continue
+			}
+			if have := st.state.GetNonce(authority); have != auth.Nonce {
+				continue
+			}
+			// If the account already exists in state, refund the new account cost
+			// charged in the initrinsic calculation.
+			if exists := st.state.Exist(authority); exists {
+				// If the account is not AccountKeyTypeLegacy, setcode is not allowed.
+				if !st.state.GetKey(authority).Type().IsLegacyAccountKey() {
+					continue
+				}
+				st.state.AddRefund(params.CallNewAccountGas - params.TxAuthTupleGas)
+			}
+			st.state.IncNonce(authority)
+			delegation := types.AddressToDelegation(auth.Address)
+			if auth.Address == common.ZeroAddress {
+				// If the delegation is for the zero address, completely clear all
+				// delegations from the account.
+				delegation = []byte{}
+			}
+			// TODO: Change to a method that adds code to EOA
```

Comment:
When TODO fix it?
I think we don't have to leave TODO comments.

Reply:
@tnasu Currently it could be inevitable to have TODOs because this feature comprises of multiple pull requests. I guess @shiki-tak will prepare another PR to fix it?

How about marking it TODO-7702 or something so we can later search for it?

Reply:
> How about marking it TODO-7702 or something so we can later search for it?

As @blukat29 says, the current plan does not allow for TODOs to be avoided. I agree with this idea.

Reply:
I'm just afraid to miss fixing it. And we don't have the power to forcibly correct it, so I think it's pointless. I believe that this part of the modification will be resolved through human communication rather than by looking at the TODOs in the source code.